psalm.xml.dist

```diff
@@ -793,10 +798,26 @@
 
                 <!-- See https://github.com/doctrine/dbal/pull/3574 -->
                 <file name="tests/Query/Expression/ExpressionBuilderTest.php"/>
-
-                <!-- See https://bugs.php.net/bug.php?id=77591 -->
-                <referencedFunction name="db2_autocommit"/>
             </errorLevel>
         </InvalidScalarArgument>
+        <MethodSignatureMustProvideReturnType>
+            <errorLevel type="suppress">
+                <!-- https://github.com/vimeo/psalm/issues/8557 -->
+                <file name="src/Query/Expression/CompositeExpression.php"/>
+                <file name="src/Query/QueryBuilder.php"/>
+            </errorLevel>
+        </MethodSignatureMustProvideReturnType>
+        <NoValue>
+            <errorLevel type="suppress">
+                <!-- We're checking for invalid input. -->
```

Comment:
That looks like a bug. You shouldn't end up with NoValue while checking types from docblock. Do you know at which line it is emitted?

Reply:
I can give you a reproducer: https://psalm.dev/r/614cec5f0f

Reply:
Thanks! I think it happens because the type is both from signature and from docblock, I'll check that

Reply:
The error occurs because I first eliminate all possible values (according to the docblock) for a variable and afterwards use its value when rendering the exception message.

Reply:
yeah, I know, the thing is that when a type comes from a docblock, when every type is refuted, Psalm assign the mixed type to it. It only applie the never type when the type came from signature. Here you have a signature but you actually have a more detailed type in docblock